rust/kcl-lib/src/execution/kcl_value.rs

```diff
@@ -58,6 +60,13 @@ pub enum KclValue {
         #[serde(rename = "__meta")]
         meta: Vec<Metadata>,
     },
+    // An array where all values have a shared type (not necessarily the same principal type).
```

Comment:
What's the different between type and principle type?

Reply:
When a type system supports subtyping, the principal type is the most specific type. So if you have an array of Animal, where Cat and Dog are subtypes of Animal, an element can have the type Animal. But its principal type would be Dog.

It's a little more complicated when the type system supports inference and generalization of type variables. I'd be happy to explain that if you're interested, but the summary is that the principal type is the type from which all other possible typings of the expression can be derived.

And yes, it's spelled "principal" even though I understand why that might be unexpected.